features/dd-sdk-android-trace/src/test/kotlin/com/datadog/trace/core/CoreSpanBuilderTest.kt

```diff
@@ -0,0 +1,534 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.trace.core
+
+import com.datadog.tools.unit.getFieldValue
+import com.datadog.trace.api.Config
+import com.datadog.trace.api.DDSpanId
+import com.datadog.trace.api.DDTags
+import com.datadog.trace.api.DDTraceId
+import com.datadog.trace.api.TracePropagationStyle
+import com.datadog.trace.api.gateway.RequestContextSlot
+import com.datadog.trace.api.naming.SpanNaming
+import com.datadog.trace.api.sampling.PrioritySampling
+import com.datadog.trace.bootstrap.instrumentation.api.AgentTracer
+import com.datadog.trace.bootstrap.instrumentation.api.AgentTracer.NoopPathwayContext
+import com.datadog.trace.bootstrap.instrumentation.api.TagContext
+import com.datadog.trace.common.writer.ListWriter
+import com.datadog.trace.core.propagation.ExtractedContext
+import com.datadog.trace.core.propagation.PropagationTags
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.AfterEach
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.Timeout
+import org.junit.jupiter.params.ParameterizedTest
+import org.junit.jupiter.params.provider.Arguments
+import org.junit.jupiter.params.provider.CsvSource
+import org.junit.jupiter.params.provider.MethodSource
+import org.mockito.kotlin.doReturn
+import org.mockito.kotlin.mock
+import java.util.concurrent.TimeUnit
+import java.util.stream.Stream
+
+@Timeout(5)
+internal class CoreSpanBuilderTest : DDCoreSpecification() {
+
+    lateinit var writer: ListWriter
+    lateinit var tracer: CoreTracer
+
+    @BeforeEach
+    override fun setup() {
+        super.setup()
+        writer = ListWriter()
+        tracer = tracerBuilder().writer(writer).build()
+    }
+
+    @AfterEach
+    override fun cleanup() {
+        super.cleanup()
+        tracer.close()
+    }
+
+    @Test
+    fun `build simple span`() {
+        val span = tracer.buildSpan(instrumentationName, "op name").withServiceName("foo").start()
+        assertThat(span.operationName).isEqualTo("op name")
+    }
```

Comment:
Let's split blocks into `Given/When/Then` here and in other tests.

Also we maybe can generate fake data as we do in other tests for the SDK (this is useful for other tests as well).

Reply:
while I do agree on the other hand this will take a lot of time to cleanup all these tests and add randomness. I am not sure we should invest this time here now as by the end of the year we might have to drop this code anyway. @plousada ?

Reply:
I don't think it takes a lot of time, but it greatly impacts the readability of the PR, because currently everything is mixed and it is hard to follow test body content tbh.

Reply:
I will take care of the `given, when, then` for now